### ISSUE TITLE
Must use the Raspbian lite image for the 0.7.0-dev branch on a Pi Zero + Explorer HAT (GUI version doesn't work)

### ISSUE THREAD

**Issue by jaylagorio** [openaps/oref0#1035]
Tl;dr: I installed this on a Pi Zero W (2018-04-18-raspbian-stretch) with the Explorer HAT. The loop log thows extensive errors with Minimed comms when I first tried to fire it up. I progressed a little closer to where I can get ``mmtune.py`` to work, but it's pretty busted and everything still throws errors. Nowhere near running a full loop.

The saga starts just after a fresh OS load:

```
sudo bash
curl -s https://raw.githubusercontent.com/openaps/oref0/dev/bin/openaps-install.sh > /tmp/openaps-install.sh && bash /tmp/openaps-install.sh
```

Hit Ctrl+C before running ``oref0-setup``

``cd ~/src/oref0 && git fetch && git checkout 0.7.0-dev && git pull && npm run global-install``

No apparent errors.

``oref0-setup --dir=/root/myopenaps --serial=xxxxxx --cgm=G5-upload --ns-host=https://xxxxx --api-secret=xxxxx --tty=/dev/spidev0.0 --enable=' autosens meal autotune ' --radio_locale='US' --btmac=xx:xx:xx:xx:xx:xx --max_iob=6.0 --current_basal_safety_multiplier=8 --max_daily_safety_multiplier=8``

No apparent errors yet here either. After a reboot though, the following log lines come flowing out as though nothing is even being run between log lines:

```
Starting oref0-pump-loop at Sun 17 Jun 15:24:08 EDT 2018 with 8 second wait_for_silence:
Waiting up to 4 minutes for new BG: First loop: not waiting
Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 15:24:22 EDT 2018
Preflight fail. Retry 1 of preflight
Preflight fail. Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 15:24:25 EDT 2018
Retry 2 of preflight
Preflight fail. Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 15:24:27 EDT 2018
Retry 3 of preflight
Preflight fail. Couldn't preflight
oref0-pump-loop failed. pump_loop_completed more than 15m old; waiting for 39 s silence before mmtuning
Could not parse profile.json:  { [Error: ENOENT: no such file or directory, open '/root/myopenaps/settings/profile.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/root/myopenaps/settings/profile.json' }
Could not parse edison-battery.json:  { [Error: ENOENT: no such file or directory, open '/root/myopenaps/monitor/edison-battery.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/root/myopenaps/monitor/edison-battery.json' }
Could not parse suggested.json:  { [Error: ENOENT: no such file or directory, open '/root/myopenaps/enact/suggested.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/root/myopenaps/enact/suggested.json' }
Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 15:24:44 EDT 2018
Listening for 39 s silence before mmtuning: Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 15:24:44 EDT 2018
mmtune: Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
No wait required.
If pump and rig are close enough, this error usually self-resolves. Stand by for the next loop.
Unsuccessful oref0-pump-loop at Sun 17 Jun 15:24:54 EDT 2018
```

Running ``mmtune.py`` yeilds an error about how the ``spi_serial`` module isn't installed. So we jump through some hoops to install that and ``mraa`` with some hints from this page (even though it's Edison-based help): https://learn.sparkfun.com/tutorials/installing-libmraa-on-ubilinux-for-edison

```
sudo apt-get install git build-essential swig3.0 python-dev nodejs-dev cmake libjson-c-dev
git clone https://github.com/intel-iot-devkit/mraa.git
mkdir mraa/build && cd $_
cmake .. -DBUILDSWIGNODE=OFF
make
make install
ldconfig && ldconfig -p | grep mraa
```

Add ``export PYTHONPATH=$PYTHONPATH:$(dirname $(find /usr/local -name mraa.py))`` to ``.bash_profile``, then install ``spi_serial``:

```
git clone https://github.com/EnhancedRadioDevices/spi_serial.git
cd spi_serial
python setup.py install
```

Now the Python module is there and works. Hooray!

```
python /usr/local/bin/mmtune.py --serial xxxxxx --port /dev/spidev0.0 --subg_rfspy --radio_locale US
{
    "scanDetails": [
        [
            "916.300",
            0,
            -99
        ],
        [
            "916.324",
            0,
            -99
        ],
        [
            "916.348",
            0,
            -99
        ],
        [
            "916.372",
            0,
            -99
        ],
        [
            "916.396",
            0,
            -99
        ],
        [
            "916.420",
            0,
            -99
        ],
        [
            "916.444",
            0,
            -99
        ],
        [
            "916.468",
            1,
            -94
        ],
        [
            "916.492",
            0,
            -99
        ],
        [
            "916.516",
            0,
            -99
        ],
        [
            "916.540",
            0,
            -99
        ],
        [
            "916.564",
            0,
            -99
        ],
        [
            "916.588",
            3,
            -88
        ],
        [
            "916.612",
            5,
            -70
        ],
        [
            "916.636",
            1,
            -93
        ],
        [
            "916.660",
            5,
            -67
        ],
        [
            "916.684",
            5,
            -67
        ],
        [
            "916.708",
            5,
            -68
        ],
        [
            "916.732",
            4,
            -78
        ],
        [
            "916.756",
            0,
            -99
        ],
        [
            "916.780",
            0,
            -99
        ],
        [
            "916.804",
            0,
            -99
        ],
        [
            "916.828",
            1,
            -94
        ],
        [
            "916.852",
            0,
            -99
        ],
        [
            "916.876",
            0,
            -99
        ]
    ],
    "setFreq": 916.684,
    "usedDefault": false
}
```

But we still have a problem with the rest of the loop (note that ``mmtune`` spits out the grep help message):

```
Starting oref0-pump-loop at Sun 17 Jun 16:01:06 EDT 2018 with 11 second wait_for_silence:
Waiting up to 4 minutes for new BG: First loop: not waiting
Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 16:01:07 EDT 2018
Preflight fail. Retry 1 of preflight
Preflight fail. Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 16:01:08 EDT 2018
Retry 2 of preflight
Preflight fail. Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 16:01:08 EDT 2018
Retry 3 of preflight
Preflight fail. Couldn't preflight
oref0-pump-loop failed. pump_loop_completed more than 15m old; waiting for 27 s silence before mmtuning
Could not parse profile.json:  { [Error: ENOENT: no such file or directory, open '/root/myopenaps/settings/profile.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/root/myopenaps/settings/profile.json' }
Could not parse edison-battery.json:  { [Error: ENOENT: no such file or directory, open '/root/myopenaps/monitor/edison-battery.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/root/myopenaps/monitor/edison-battery.json' }
Could not parse suggested.json:  { [Error: ENOENT: no such file or directory, open '/root/myopenaps/enact/suggested.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/root/myopenaps/enact/suggested.json' }
Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 16:01:17 EDT 2018
Listening for 27 s silence before mmtuning: Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 16:01:18 EDT 2018
mmtune: Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
No wait required.
If pump and rig are close enough, this error usually self-resolves. Stand by for the next loop.
Unsuccessful oref0-pump-loop at Sun 17 Jun 16:01:27 EDT 2018
```

**Comment by cluckj** [comment]
The python-based pump comms have mostly been replaced in 0.7.0, but I think the installer didn't do what it was supposed to. Did you run the oref0-setup.sh in interactive mode?

**Comment by cluckj** [comment]
Yeah, there are a couple new setup parameters that 0.7.0 introduces, and some code that auto-detects the explorer HAT in the interactive installer.

**Comment by jaylagorio** [comment]
OK, this is the recommended "run again with the same options" command line from interactive setup (close enough):

``oref0-setup --dir=/root/myopenaps --serial=386591 --cgm=g5-upload --ns-host=http://nightscout.lagor.io --api-secret=THISISTHESECRET --tty=/dev/spidev0.0 --max_iob=6 --enable=' autotune ' --radio_locale='US' --btmac='CC:F3:A5:DA:AD:7B'``

I noticed that the automated setup that I originally ran did detect the Explorer HAT and made mention of it, but I would expect the automated setup to perform the same way as the interactive setup with the same options selected.

I should add that when it didn't work initially I went to my go-to method of debugging comms issues, `mmtune.py`. I knew that parts of the comms were switching, but since the script was still there I expected it to work. Can we have a commit that removes all the obsolete scripts we don't expect to work or don't want people to use so we can see what breaks before this goes to dev?

**Comment by cluckj** [comment]
mmeowlink isn't entirely obsolete (it's still needed for Enlite CGM), so we can't remove it yet. The SPI serial and MRAA dependencies do get installed when someone selects mdt as their cgm.

Can you try a fresh reinstall of Raspbian, followed by the branch switch & interactive setup?

**Comment by jaylagorio** [comment]
New clean OS install, new branch switch, new interactive setup:

```
Continuing oref0-pump-loop at Sun 17 Jun 20:15:21 EDT 2018
Listening for 5 s silence before mmtuning: Listening: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Sun 17 Jun 20:15:21 EDT 2018
mmtune: Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
No wait required.
If pump and rig are close enough, this error usually self-resolves. Stand by for the next loop.
Unsuccessful oref0-pump-loop at Sun 17 Jun 20:15:48 EDT 2018
```

Let me know what you want me to do to work this out.

**Comment by scottleibrand** [comment]
You can manually run mmtune the way that oref0-pump-loop now runs it by doing:

```
cd ~/myopenaps/
export MEDTRONIC_PUMP_ID=`grep serial pump.ini | tr -cd 0-9`
killall -g oref0-pump-loop
Go-mmtune
```

**Comment by cluckj** [comment]
Those commands going by quickly is symptomatic of spi not being enabled (they will show up in the logs between the initial setup and the reboot). If that's still happening after reboot, check to see if `dtparam=spi=on` is set in `/boot/config.txt`?

**Comment by jaylagorio** [comment]
@scottleibrand Thanks for the new ``mmtune`` method@

@cluckj That setting is present, the first thing I do after booting the OS to the GUI is connecting to wifi and using the ``raspi-config`` interface to set autologin, boot-to-console, and enable all the interfaces, then reboot.

I just rebooted with ``cron`` disabled and tried the new ``mmtune`` command:

```
2018/06/18 06:34:26 cannot connect to CC111x radio on /dev/spidev0.0
2018/06/18 06:34:26 write /sys/class/gpio/export: device or resource busy
```

What else can I try?

**Comment by scottleibrand** [comment]
Sounds like you’re not following the docs if you’re using the Raspberry Pi GUI. Not sure if that matters, but you might try again with the lite (non-GUI) image.

**Comment by jaylagorio** [comment]
@scottleibrand I'll give you a very confused round of applause for pointing out the lite image being the culprit - I don't know what *isn't* there in the full OS image that is there in lite, but doing everything above the same except for the OS version seems to have gotten me moving forward! Mind if I keep this issue open until I get a full loop going? I'd like to add xdrip-js with its different version of node to see if everything keeps working correctly.

**Comment by scottleibrand** [comment]
Good to know.  If you or anyone else cares to try to figure out why the full GUI image won't work, we might be able to make it work.  Until then, we should probably do a docs PR to note that only the Lite image works for now.

I'm fine keeping the issue open, but please retitle it to reflect what we found.

**Comment by jaylagorio** [comment]
Done! Thank you!

**Comment by jaylagorio** [comment]
Quick PR in `openaps-menu`: https://github.com/openaps/openaps-menu/pull/15

**Comment by jaylagorio** [comment]
Full loop works now, even after you upgrade the version of node with the directions from Logger (https://github.com/xdrip-js/Logger):

```
mkdir node && cd node
wget https://nodejs.org/dist/v8.10.0/node-v8.10.0-linux-armv6l.tar.xz
tar -xf node-v8.10.0-linux-armv6l.tar.xz
cd *6l
sudo cp -R * /usr/local/
# add /usr/local/bin to PATH env variable in .bash_profile
node -v
# should show 8.10.0 at this point
```

`openaps-menu` complains about being compiled by a different version of node (and breaks the loop) if you do the upgrade after `oref0-setup`. Is there a chance we can get 0.7.0 to install this version of node in time for your release window so Logger is just a `git clone` away from working?

**Comment by scottleibrand** [comment]
Is this a fresh install or an upgrade? I think we install node 8 if there’s no node yet.

**Comment by jaylagorio** [comment]
It was a fresh OS, then the OpenAPS setup script and `oref0-setup`, and then Logger was added after that. Logger didn't work until I did the upgrade steps listed above.

**Comment by scottleibrand** [comment]
Which node version did you have before upgrading?

**Comment by cluckj** [comment]
I have v4.8.2 on my latest 4/18 Raspbian install from a couple days ago.

(btw, you can run `sudo npm install` in `~/src/openaps-menu/` and it'll rebuild it with the new version of node.)

**Comment by scottleibrand** [comment]
So should we do https://github.com/openaps/oref0/commit/9006a6d831095beddeb0774cdc074ce37573ec98 ?

**Comment by jaylagorio** [comment]
That's the one! Once you do the update for Logger `node -v` reports `v8.10.0`. And yes, that's the command I ran when I saw the version mismatch errors in the loop log.

**Comment by jaylagorio** [comment]
@scottleibrand Do you want to put https://github.com/openaps/oref0/commit/9006a6d831095beddeb0774cdc074ce37573ec98 into 0.7.0-dev and I'll try a clean install?

**Comment by scottleibrand** [comment]
Want to just try the entire misc-stuff branch? We’re running it without issue and have a PR open for it “the post-refactor one”.

**Comment by jaylagorio** [comment]
Sorry, switch to that branch and try it? Because if you merged it here I would update and try it tonight too.

**Comment by jaylagorio** [comment]
I'm not the most git-oriented person, so I just caught that `misc-stuff` is rooted on `0.7.0-dev` so it doesn't actually have to be merged here in order to test `0.7.0-dev` plus those changes. I'll try to test that out later today.

**Comment by jaylagorio** [comment]
Still no major issues, but after a couples days of watching the loop log next to Nightscout I've noticed some things where what the loop has for data isn't always consistent with what's being displayed in Nightscout. The time delay isn't very long of course, but when I look at Nightscout (and it's not in the middle of a running loop) I expect it to reflect the state of the loop at the end of its last execution:

 - The OpenAPS pill reports a last runtime of around 4 min even at the end of a loop (and around 9 min around when the loop starts again). It never seems to get below that. When "completed oref0-pump-loop" gets logged the time in the pill is at that 4 minute mark.

 - After a loop-initiated bolus the new boluses don't show up on Nightscout for at least one iteration of the loop. I think this is a regression - in a previous discussion a while ago (on Gitter?) the loop-initiated boluses with new IOB calculation were sent up to Nightscout nearly immediately.

 - I think this is related to the above item, but I've noticed that a complete loop will finish with "bolusing" being the status when it gives boluses (and as mentioned the bolus doesn't show up until the start of the next loop). Could we add a wait loop where if the status from the pump is bolusing it waits 5 - 10 seconds, polls again, and continues to poll until until the status is error or normal where it then continues? Or is this intended behavior? If I remember what you said from our previous discussion this might also involve another read of pump events to see the completed bolus and send it up to Nightscout.

**Comment by scottleibrand** [comment]
There are power saving optimizations that avoid re-doing pumphistory pulls and IOB recalculations unless you’re on wall power. You may get more prompt uploads if you plug in the rig.

**Comment by jaylagorio** [comment]
How are you determining whether it's plugged into power? Since I was home-bound today and yesterday I was plugged in the entire time with the battery pill consistently reporting > 95%. I don't know that I think power savings is more important than reporting accuracy.

**Comment by scottleibrand** [comment]
I’d take a look at oref0-pump-loop.sh and compare it to your pump-loop.log output.

**Comment by jaylagorio** [comment]
Thanks for the pointer, I think I found it! To correct the above I'd recommend the following changes:

 - In `main` (https://github.com/openaps/oref0/blob/f754421f8e0189e9877c2b15a97a2789e2eb3854/bin/oref0-pump-loop.sh#L79) remove the `onbattery` call. If this does what I think it does it would bring the Nightscout display current regardless of charging status and I think that's important.
 - In `onbattery` (https://github.com/openaps/oref0/blob/f754421f8e0189e9877c2b15a97a2789e2eb3854/bin/oref0-pump-loop.sh#L819) drop the threshold to 95% at least, 90% maybe to be safe?

**Comment by scottleibrand** [comment]
The 90% threshold seems like a no-real-downside change: I’d be happy to merge a PR there. Not sure we want to turn off power savings entirely, though, given how power hungry the Pi is compared to the Edison.

**Comment by jaylagorio** [comment]
Just filed https://github.com/openaps/oref0/pull/1050. Funny that an ARM-based board is more power hungry than an x86-based board! That's definitely not something I would have assumed, is that still true to comparing an underclocked Pi Zero to an Edison with standard settings?

I'm not proposing removing power savings altogether, although I haven't looked at where else `onbattery` calls could be sprinkled to achieve more. Would it help to alter `refresh_profile` to only call `get_settings` (which does a bunch of pump interaction) when given a certain time period? I can try and look for some other places to use `onbattery` but I really think getting accurate data to the user in near real-time is important. The inconsistency of having pump-loop.log differ from Nightscout presents doubt.

**Comment by tuzoenduro** [comment]
I get the same errors, even though I did a proper raspbian lite install (I have tried both with PiBakery and with Etcher)... and no successful loops... I posted on gitter but have had no luck getting this working.. I will paste a copy of my first (NOK) loop :
`Starting oref0-pump-loop at Tue 22 Jan 11:24:15 CET 2019 with 13 second wait_for_silence:
Waiting up to 4 minutes for new BG: First loop: not waiting

Listening for 13s: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Tue 22 Jan 11:24:32 CET 2019
Preflight OK. Old settings: jq: error (at settings/pumpprofile.json.new:1): Cannot index number with string "current_basal"
Invalid pumpprofile.json.new after refresh
-rw-r--r-- 1 root root 3 Jan 22 11:25 settings/pumpprofile.json.new
Could not parse autotune_data
Could not parse temptargets_data.
ERROR: bad basal schedule [ { i: 0, start: '00:00:00', minutes: 0, rate: 0 } ]
max_daily_basal of 0 is not supported
jq: error (at settings/profile.json.new:1): Cannot index number with string "current_basal"
Invalid profile.json.new after refresh
-rw-r--r-- 1 root root 3 Jan 22 11:25 settings/profile.json.new
Profile valid. Pump history updated through 2019-01-22T11:04:43+01:00 with 0 new records; meal.json Could not parse input data:  { Error: ENOENT: no such file or directory, open 'settings/profile.json'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-meal.js:50:42)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'settings/profile.json' }
{ "carbs": 0, "mealCOB": 0, "reason": "Could not parse input data" }
cat: monitor/meal.json: No such file or directory
Couldn't check_cp_meal - continuing
Retry 1 of refresh_pumphistory_and_meal
Pump history updated through 2019-01-22T11:04:43+01:00 with 0 new records; meal.json Could not parse input data:  { Error: ENOENT: no such file or directory, open 'settings/profile.json'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-meal.js:50:42)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'settings/profile.json' }
{ "carbs": 0, "mealCOB": 0, "reason": "Could not parse input data" }
cat: monitor/meal.json: No such file or directory
Couldn't check_cp_meal - continuing
Listening for 7s: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Tue 22 Jan 11:28:23 CET 2019
Retry 2 of refresh_pumphistory_and_meal
Pump history updated through 2019-01-22T11:04:43+01:00 with 0 new records; meal.json Could not parse input data:  { Error: ENOENT: no such file or directory, open 'settings/profile.json'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-meal.js:50:42)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'settings/profile.json' }
{ "carbs": 0, "mealCOB": 0, "reason": "Could not parse input data" }
cat: monitor/meal.json: No such file or director
Couldn't check_cp_meal - continuing
Listening for 13s: .No interfering pump comms detected from other rigs (this is a good thing!)
Continuing oref0-pump-loop at Tue 22 Jan 11:29:58 CET 2019
Retry 3 of refresh_pumphistory_and_meal
Pump history updated through 2019-01-22T11:04:43+01:00 with 0 new records; meal.json Could not parse input data:  { Error: ENOENT: no such file or directory, open 'settings/profile.json'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-meal.js:50:42)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'settings/profile.json' }
{ "carbs": 0, "mealCOB": 0, "reason": "Could not parse input data" }
cat: monitor/meal.json: No such file or directory
Couldn't check_cp_meal - continuing
Couldn't refresh_pumphistory_and_meal
oref0-pump-loop failed. pump_loop_completed more than 15m old; waiting for 38 s silence before mmtuning
Updating HAT Display...`

I also get an error while running the setup script 

`npm ERR! invalid: oref0@0.7.0-dev /usr/local/lib/node_modules/oref0
npm ERR! not ok code 0`

**Comment by scottleibrand** [comment]
Looks like you have a pump basal of zero at midnight? That isn’t allowed. Update that and try again?

**Comment by tuzoenduro** [comment]
Scott, i got it sorted. I come from a 640, and did not know the 722 has a preferred basal. I had imput all my values onto the ‘pattern a’ profile and had let the standard one at 0.

I copied it all and we’re in business now, thank you for the help.

Dr. luis toussaint
________________________________

**Comment by danamlewis** [comment]
I think all the issues within this issue have been resolved. Please open a new issue if there is a related issue on the latest 0.7.0-dev release candidate (in October 2019).
